.github/workflows/ci.yml

```diff
@@ -1,51 +1,54 @@
 name: CI
-on: [push, pull_request]
+
+on:
+  - push
+  - pull_request
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
 
   lint:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/cache@v2
+      - uses: actions/checkout@v4
+      - uses: actions/cache@v4
         with:
           path: '**/node_modules'
           key: ${{ runner.os }}-lint-modules-${{ hashFiles('**/yarn.lock') }}
-      - uses: actions/setup-node@v2
+      - uses: actions/setup-node@v4
         with:
-          node-version: 14.x
-      - run: yarn install
+          node-version: lts/*
+      - run: yarn install --frozen-lockfile
       - run: yarn run lint
 
   test:
+    needs: lint
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os: [ubuntu-latest]
         node-version:
-          - 14.x
-          - 16.x
-          - 18.x
+          - lts/-3
```

Comment:
Not sure I'm a fan of this change.
This would mean that our CI could suddenly start failing upon new Node releases without us changing our code.
I would prefer our CI to be fully deterministic.

Reply:
Okay! I thought it would be handy to have it automatically update, to remove the need to manually adjust the versions every time. I will change this back.